layers/core_checks/cc_data_graph.cpp

```diff
@@ -0,0 +1,448 @@
+    if (skip) {
+        return true;
+    }
+
+    if ((pCreateInfo->flags & VK_DATA_GRAPH_PIPELINE_SESSION_CREATE_PROTECTED_BIT_ARM) != 0) {
+        if (!enabled_features.protectedMemory) {
+            skip |= LogError(
+                "VUID-VkDataGraphPipelineSessionCreateInfoARM-protectedMemory-09782", device, pipeline_loc.dot(Field::flags),
+                "(%s) contains VK_DATA_GRAPH_PIPELINE_SESSION_CREATE_PROTECTED_BIT_ARM but the protectedMemory feature is not enabled",
+                string_VkDataGraphPipelineSessionCreateFlagsARM(pCreateInfo->flags).c_str());
+        }
+    }
+
+    return skip;
+}
+
+bool CoreChecks::PreCallValidateGetDataGraphPipelineSessionBindPointRequirementsARM(VkDevice device, const VkDataGraphPipelineSessionBindPointRequirementsInfoARM* pInfo, uint32_t* pBindPointRequirementCount, VkDataGraphPipelineSessionBindPointRequirementARM* pBindPointRequirements, const ErrorObject& error_obj) const {
+    bool skip = false;
+    const auto session_ptr = Get<vvl::DataGraphPipelineSession>(pInfo->session);
+    const Location pipeline_loc = error_obj.location.dot(Field::dataGraphPipeline);
+    if(!session_ptr) {
+        skip |= LogError("VUID-vkGetDataGraphPipelineSessionBindPointRequirementsARM-session-09783", device, pipeline_loc, "The session member of pInfo must have been created with device.");
+    }
+    return skip;
+
+}
+
+void CoreChecks::PostCallRecordGetDataGraphPipelineSessionBindPointRequirementsARM(
+    VkDevice device, const VkDataGraphPipelineSessionBindPointRequirementsInfoARM* pInfo, uint32_t* pBindPointRequirementCount,
+    VkDataGraphPipelineSessionBindPointRequirementARM* pBindPointRequirements, const RecordObject& record_obj) {
+    if (VK_SUCCESS != record_obj.result) {
+        return;
+    }
+    if (auto session_ptr = Get<vvl::DataGraphPipelineSession>(pInfo->session)){
+        if (pBindPointRequirements) {
+            session_ptr->InitMemoryRequirements(device, pBindPointRequirements, *pBindPointRequirementCount);
+        }
+    }
+}
+
+bool CoreChecks::PreCallValidateGetDataGraphPipelineSessionMemoryRequirementsARM(
+    VkDevice device, const VkDataGraphPipelineSessionMemoryRequirementsInfoARM* pInfo, VkMemoryRequirements2* pMemoryRequirements,
+    const ErrorObject& error_obj) const {
+    bool skip = false;
+    auto session_ptr = Get<vvl::DataGraphPipelineSession>(pInfo->session);
+    ASSERT_AND_RETURN_SKIP(session_ptr);
+    auto& session = *session_ptr;
+    const Location pinfo_loc = error_obj.location.dot(Field::pInfo);
+    if (auto bpr = session.FindBindPointRequirement(pInfo->bindPoint)) {
+        if (VK_DATA_GRAPH_PIPELINE_SESSION_BIND_POINT_TYPE_MEMORY_ARM != bpr->bindPointType) {
+            skip |= LogError(
+                "VUID-vkGetDataGraphPipelineSessionMemoryRequirementsARM-bindPoint-09784", device, pinfo_loc.dot(Field::bindPoint),
+                "bindPoint %s has type %s, expected VK_DATA_GRAPH_PIPELINE_SESSION_BIND_POINT_TYPE_MEMORY_ARM",
```

Comment:
```suggestion
                "%s has type %s, expected VK_DATA_GRAPH_PIPELINE_SESSION_BIND_POINT_TYPE_MEMORY_ARM",
```

same with rest of this function